storagemarket/helper.go

```diff
@@ -0,0 +1,180 @@
+package storagemarket
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/api/v1api"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/market"
+	ctypes "github.com/filecoin-project/lotus/chain/types"
+
+	"github.com/filecoin-project/go-fil-markets/storagemarket"
+	"github.com/filecoin-project/go-state-types/exitcode"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/build"
+	market2 "github.com/filecoin-project/specs-actors/v2/actors/builtin/market"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+)
+
+type MinerHelper struct {
```

Comment:
nit: can we use a more descriptive name than MinerHelper? IIUC the only method that gets mocked out is WaitForPublishDeals, so maybe it should be combined with the DealPublish interface, or it could be called something like ChainDealManager